components/taxonomy/src/values.rs

```diff
@@ -572,6 +577,143 @@ impl Deserialize for DoorLocked {
     }
 }
 
+/// A secure/insecure state.
+///
+/// # JSON
+///
+/// This kind is represented by strings "Secure" | "Insecure".
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum Secure {
+    /// # JSON
+    ///
+    /// Represented by "Insecure".
+    ///
+    /// ```
+    /// use foxbox_taxonomy::values::*;
+    /// use foxbox_taxonomy::parse::*;
+    ///
+    /// let parsed = Secure::from_str("\"Insecure\"").unwrap();
+    /// assert_eq!(parsed, Secure::Insecure);
+    ///
+    /// let serialized: JSON = Secure::Insecure.to_json();
+    /// assert_eq!(serialized.as_string().unwrap(), "Insecure");
+    /// ```
+    Insecure,
+
+    /// # JSON
+    ///
+    /// Represented by "Secure".
+    ///
+    /// ```
+    /// use foxbox_taxonomy::values::*;
+    /// use foxbox_taxonomy::parse::*;
+    ///
+    /// let parsed = Secure::from_str("\"Secure\"").unwrap();
+    /// assert_eq!(parsed, Secure::Secure);
+    ///
+    /// let serialized: JSON = Secure::Secure.to_json();
+    /// assert_eq!(serialized.as_string().unwrap(), "Secure");
+    /// ```
+    Secure,
+}
+
+impl Secure {
+    fn as_bool(&self) -> bool {
+        match *self {
+            Secure::Insecure => false,
+            Secure::Secure => true,
+        }
+    }
+}
+
+impl Parser<Secure> for Secure {
+    fn description() -> String {
+        "Secure".to_string()
+    }
+    fn parse(path: Path, source: &mut JSON) -> Result<Self, ParseError> {
+        match source.as_string() {
+            Some("Insecure") => Ok(Secure::Insecure),
+            Some("Secure") => Ok(Secure::Secure),
+            Some(str) => Err(ParseError::unknown_constant(str, &path)),
+            None => Err(ParseError::type_error("Secure", &path, "string"))
+        }
+    }
+}
+
+impl ToJSON for Secure {
+    fn to_json(&self) -> JSON {
+        match *self {
+            Secure::Insecure => JSON::String("Insecure".to_owned()),
+            Secure::Secure => JSON::String("Secure".to_owned())
+        }
+    }
+}
+impl Into<Value> for Secure {
+    fn into(self) -> Value {
+        Value::Secure(self)
+    }
+}
+
+impl PartialOrd for Secure {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Secure {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.as_bool().cmp(&other.as_bool())
+    }
+}
+
+///
```

Comment:
as said on IRC, Value itself derive Serialize and Deserialize, so I need them for now.